ntp-proto/src/timetypes.rs

```diff
@@ -0,0 +1,45 @@
+#[derive(Debug, Copy, Clone, Eq, PartialEq, PartialOrd, Ord)]
+pub struct NtpTimestamp {
+    timestamp: u64,
+}
+
+impl NtpTimestamp {
+    pub(crate) fn from_bits(bits: [u8; 8]) -> NtpTimestamp {
```

Comment:
Style nit: I prefer to use `Self` pervasively in return types and struct initializers. I like to `warn(clippy::use_self)` for this.